src/vos/evtree.c

```diff
@@ -2568,14 +2589,11 @@ evt_ent_array_fill(struct evt_context *tcx, enum evt_find_opc find_opc,
 				if (rc < 0)
 					D_GOTO(out, rc);
 
-				if (range_overlap == RT_OVERLAP_INCLUDED &&
-				    rect->rc_minor_epc == EVT_MINOR_EPC_MAX) {
-					D_ERROR("Ignore RT_OVERLAP_INCLUDED array remove "
-						DF_RECT" and "DF_RECT"\n", DP_RECT(rect),
-						DP_RECT(&rtmp));
-					rc = 0;
-					goto out;
-				}
+				D_ASSERTF(rect->rc_minor_epc != EVT_MINOR_EPC_MAX,
+					  "Should never have overlap with removals: " DF_RECT
+					  " overlaps with " DF_RECT "\n",
+					  DP_RECT(&rtmp), DP_RECT(rect));
+
```

Comment:
We regard that caller inserts overlapped removal records as bug, and this assert is to capture the bug?

Reply:
Yes.  The only way to insert removal records is through evt_remove_all and this should never insert overlapped records.  The code here was added by Xuezhao to work around the bug I'm fixing with this patch.